chemclipse/plugins/org.eclipse.chemclipse.chromatogram.xxd.filter.supplier.savitzkygolay/src/org/eclipse/chemclipse/chromatogram/xxd/filter/supplier/savitzkygolay/processor/SavitzkyGolayProcessor.java

```diff
@@ -385,4 +386,25 @@ public IChromatogramFilterResult apply(ITotalScanSignals totalSignals, Chromatog
 		}
 		return new ChromatogramFilterResult(ResultStatus.OK, "The Savitzky-Golay filter has been applied successfully.");
 	}
+	
+	public IChromatogramFilterResult apply(ExtractedMatrix extractedMatrix, ChromatogramFilterSettings filterSettings, IProgressMonitor monitor) {
```

Comment:
It seems like this method could be made static, it would be even better if the method itself only accepts an double[][] matrix as a parameter and does return void.
Calling code can the use extractedmatrix to extract/writeback and if required create any returnvalue it feels suitable